tests/test_docker.py

```diff
@@ -88,7 +88,45 @@ def test_instance_activate_need_pull_image(agent, responses):
 def test_instance_only_activate(agent, responses):
     _delete_container('/c861f990-4472-4fa1-960f-65171b544c28')
 
+    def pre(req):
+        instance = req['data']['instanceHostMap']['instance']
+        for nic in instance['nics']:
+            nic['macAddress'] = ''
+
+    def post(req, resp):
+        docker_inspect = resp['data']['instance']['+data']['dockerInspect']
+        mac_received = docker_inspect['Config']['MacAddress']
+        mac_nic_received = docker_inspect['NetworkSettings']['MacAddress']
+        assert mac_received == ''
+        assert mac_nic_received is not None
+        del resp['data']['instance']['+data']['dockerInspect']
+        docker_container = resp['data']['instance']['+data']['dockerContainer']
+        fields = resp['data']['instance']['+data']['+fields']
+        del docker_container['Created']
+        del docker_container['Id']
+        del docker_container['Status']
+        docker_container = _sort_ports(docker_container)
+        del docker_container['Ports'][0]['PublicPort']
+        del docker_container['Ports'][1]['PublicPort']
+        del fields['dockerIp']
+        assert fields['dockerPorts']['8080/tcp'] is not None
+        assert fields['dockerPorts']['12201/udp'] is not None
+        fields['dockerPorts']['8080/tcp'] = '1234'
+        fields['dockerPorts']['12201/udp'] = '5678'
+
+    event_test(agent, 'docker/instance_activate', pre_func=pre, post_func=post)
+
+
+@if_docker
+def test_instance_only_activate_mac_address(agent, responses):
+    _delete_container('/c861f990-4472-4fa1-960f-65171b544c28')
+
     def post(req, resp):
+        docker_inspect = resp['data']['instance']['+data']['dockerInspect']
+        mac_received = docker_inspect['Config']['MacAddress']
+        mac_nic_received = docker_inspect['NetworkSettings']['MacAddress']
+        assert mac_received == '02:03:04:05:06:07'
+        assert mac_nic_received is not None
         del resp['data']['instance']['+data']['dockerInspect']
         docker_container = resp['data']['instance']['+data']['dockerContainer']
         fields = resp['data']['instance']['+data']['+fields']
```

Comment:
Actually, all off my comments apply down here but I can't apply comments on the lines in your function not up there.
The diff tool is interpreting the change...in an *interesting* way.